src/signal_manager.sh

```diff
@@ -0,0 +1,48 @@
+. "$KW_LIB_DIR/kwio.sh" --source-only
+
+declare -ga SIGNAL_MANAGER_VALID_SIGNALS=( $(trap -l | sed -E 's/[0-9]+\)/ /g') )
+
+# This functions prints the default message when interrupting kw
+function default_interrupt_handler()
+{
+  say $'\n'Oh no! An interruption! See ya...
+}
+
+# This function adds a new signal handler to an arbitrary signal
+#
+# @command The handler to be called when a signal is catched
+# @signals The signals to attach to this handler
+#
+# If command is empty, use the default handler
+# If the signal is empty, use the default signals
+
+function signal_manager()
+{
+  local command="$1"
+  shift
+  local -a signals=( "$@" )
+
+  if [[ -z "$command" ]]; then
+    command='default_interrupt_handler'
+  fi
+
+  if [[ "${#signals}" = 0 ]]; then
+    signals=( SIGINT SIGTERM )
+  else
+    for s in "${signals[@]}"; do
```

Comment:
Is this loop working as you expect? I mean, from `"${signals[@]}"` and the lack of `IF=...` I suspect that this loop only runs a single time no matter how many parameters you have. You can check it by adding a simple `echo "---> $s" inside this loop and ensure that you pass multiple signals.

Reply:
my [other](https://github.com/kworkflow/kworkflow/pull/282#discussion_r643485049) comment addresses this issue!